provider/maas/maas2instance.go

```diff
@@ -54,10 +55,25 @@ func (mi *maas2Instance) Id() instance.Id {
 }
 
 func (mi *maas2Instance) Addresses() ([]network.Address, error) {
-	machineAddresses := mi.machine.IPAddresses()
-	addresses := make([]network.Address, len(machineAddresses))
-	for i, address := range machineAddresses {
-		addresses[i] = network.NewAddress(address)
+	subnetsMap, err := mi.environ.subnetToSpaceIds()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	// Get all the interface details and extract the addresses.
+	interfaces, err := maas2NetworkInterfaces(mi, subnetsMap)
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var addresses []network.Address
+	for _, iface := range interfaces {
+		if iface.Address.Value != "" {
+			logger.Debugf("found address %q on interface %q", iface.Address, iface.InterfaceName)
+			addresses = append(addresses, iface.Address)
+		} else {
+			logger.Infof("no address found on interface %q", iface.InterfaceName)
```

Comment:
Info also seems uninteresting for a sysadmin who is just watching the logs (the threshold should be what things trigger occasionally that an operator would be interested in.)
We want to have the default log level be INFO and have it be useful.